src/devices/OneWire/samples/OneWire.Sample.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Iot.Device.OneWire.Samples
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            // Make sure you can access the bus device before requesting a device scan
+            // $ sudo chmod a+rw /sys/bus/w1/devices/w1_bus_master1/w1_master_*
+            foreach (var bus in OneWireController.EnumerateBuses())
+            {
+                Console.WriteLine($"Found bus '{bus.DeviceId}', scanning for devices ...");
+                await bus.ScanForDevicesAsync();
```

Comment:
should EnumerateDevices call the scanner?

Reply:
No, scan is an expensive operation. If you just want to enumerate devices already found, this is fast. Scan is when you hot plug devices.